src/tentaclio/clients/databricks_client.py

```diff
@@ -0,0 +1,99 @@
+
+Extends SQLAlchemyClient adding methods to get simba driver and http_path.
+"""
+
+import os
+import platform
+import urllib
+from typing import Union
+
+from sqlalchemy.engine import Connection, create_engine, result
+
+from tentaclio import urls
+
+from . import decorators, sqla_client
+
+
+__all__ = ["DatabricksClient"]
+
+SIMBA_DRIVER_DEFAULT_PATHS = {
+    "Windows": "path/to/simba/",
+    "Darwin": "/Library/simba/spark/lib/libsparkodbc_sbu.dylib",
+    "Linux": "/opt/simba/spark/lib/64/libsparkodbc_sb64.so",
+}
+
+
+class DatabricksClient(sqla_client.SQLAlchemyClient):
+    """Databricks client, backed by a pyodbc + SQLAlchemy connection."""
+
+    def __init__(self, url: Union[str, urls.URL]) -> None:
+        """Create sqlalchemy client based on the passed url.
+
+        This is a wrapper for sqlalchemy engine/connection creation.
+        """
+        super().__init__(url)
+        self.driver = _get_simba_driver_path()
+        self.http_path = self._get_http_path()
+
+    def _get_http_path(self) -> str:
+        http_path = os.environ.get("TENTACLIO__DATABRICKS_HTTP_PATH", "")
+        if http_path == "" and self.url.query:
+            http_path = self.url.query["http_path"]
+        return http_path
+
+    def _connect(self) -> Connection:
+        connection_url = _build_odbc_connection_string(
+            DRIVER=self.driver,
+            HOST=self.host,
+            PORT=self.port,
+            UID=self.username,
+            PWD=self.password,
+            Schema=self.database,
+            HTTPPath=self.http_path,
+            AuthMech=3,
+            SparkServerType=3,
+            ThriftTransport=2,
+            SSL=1,
+            IgnoreTransactions=1,
+        )
+
+        if self.engine is None:
+            self.engine = create_engine(
+                f"mssql+pyodbc:///?odbc_connect={connection_url}"
+            )
+        return self.engine.connect()
+
+    # Query methods:
+
+    @decorators.check_conn
+    def query(self, sql_query: str, **kwargs) -> result.ResultProxy:
+        """Execute a read-only SQL query, and return results.
+
+        This will not commit any changes to the database.
+        """
+        return self.conn.execute(sql_query, **kwargs)
+
+    @decorators.check_conn
+    def execute(self, sql_query: str, **kwargs) -> None:
+        """Execute a raw SQL query command."""
+        trans = self.conn.begin()
+        try:
+            self.conn.execute(sql_query, **kwargs)
+        except Exception:
+            trans.rollback()
+            raise
+        else:
+            trans.commit()
+
+
+def _build_odbc_connection_string(**kwargs) -> str:
+    connection_url = ";".join([f"{k}={v}" for k, v in kwargs.items()])
+    return urllib.parse.quote(connection_url)
+
+
+def _get_simba_driver_path() -> str:
+    driver_path = os.environ.get("TENTACLIO__SIMBA_DRIVER_PATH", "")
+    if driver_path == "":
+        driver_path = SIMBA_DRIVER_DEFAULT_PATHS[platform.system()]
+    return driver_path
```

Comment:
I think it'd be better to try to keep as much of the configuration in the tentaclio secrets file as possible rather than using exclusively env vars - perhaps we could include the driver path as a URL query? We could do the same thing as we do for postgres application name and as you do for the HTTPPath, where you _can_ specify it as an env var but it'll fall back to the URL query if it's not there